src/gt4py/next/iterator/type_inference.py

```diff
@@ -439,7 +559,7 @@ def visit_SymRef(self, node: ir.SymRef, *, symtable, **kwargs) -> Type:
         if node.id in (ir.BUILTINS - ir.TYPEBUILTINS):
             raise NotImplementedError(f"Missing type definition for builtin '{node.id}'")
 
-        return TypeVar.fresh()
+        return TypeVar.fresh()  # TODO document when this case (LetPolymorphic?)
```

Comment:
The simplest case for this path is quite trivial: it is required for type inference of the basic ITIR expression consisting of just a single symbol (e.g., `itir.SymRef(id="x")`).

Reply:
I think I added this comment, because this default case (lookup in symtable failed) can be reached if you forget to pass a symtable (e.g. by not propagating in visits). I don't remember now the exact case. I guess type inference should also work for symbols that are not defined (in the current subtree), therefore probably it cannot be improved.

Reply:
removed the comment...